tail/tail.go

```diff
@@ -191,12 +191,25 @@ func tailSingleFile(tailer *tail.Tail, file string, stateFile string) chan strin
 	}()
 
 	go func() {
-		for line := range tailer.Lines {
-			if line.Err != nil {
-				// skip errored lines
-				continue
+	ReadLines:
+		for {
+			select {
+			case line, ok := <-tailer.Lines:
+				if !ok {
+					// tailer.Lines is closed
+					break ReadLines
+				}
+				if line.Err != nil {
+					// skip errored lines
+					continue
+				}
+				lines <- strings.TrimSpace(line.Text)
+			case <-abort:
+				break ReadLines
+			default:
+				// no new lines to process, just chill for a bit
+				time.Sleep(50 * time.Millisecond)
```

Comment:
yes - if it's a quiet log file and there are no updates we don't want to spin on trying to read it because that eats up CPU wastefully.